src/lighteval/data.py

```diff
@@ -132,7 +133,7 @@ def splits_start_end_iterator(self) -> tuple[int, int]:
         Yields:
             tuple: A tuple containing the start and end indices of a split.
         """
-        for split_id in range(self.num_dataset_splits):
+        for split_id in range(self.total_size and self.num_dataset_splits):
```

Comment:
This adds complexity and makes the code less legible - it's a nice trick but it would be better to do something like
```python
split_range = self.num_dataset_split
if self.total_size == 0:
    split_range = 0 # or even simply return
```

Reply:
You also need to edit the doc to explain the docstring edit

Reply:
> You also need to edit the doc to explain the docstring edit

What did you mean by this? I couldn't get that.

Reply:
Sorry, I meant edit the docstring to explain the function signature edit
(since now the signature is `def splits_start_end_iterator(self) -> Generator[Tuple[int, int], None, None]:`)

Reply:
I changed to a yet correct annotation, removing the need to change docstring.